### PR TITLE
Auto reload segments when schema or table config updated

### PR DESCRIPTION
**Pull request by jackjlli**
This PR adds code to enable auto reload segments when schema/table config is updated.
When schema is changed, since schema name and raw table name should be the same, we fetch existing table names by schema name.

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java [hunk 1]
```diff
@@ -115,8 +123,9 @@ public SuccessResponse deleteSchema(
   @ApiResponses(value = {@ApiResponse(code = 200, message = "Successfully updated schema"), @ApiResponse(code = 404, message = "Schema not found"), @ApiResponse(code = 400, message = "Missing or invalid request body"), @ApiResponse(code = 500, message = "Internal error")})
   public SuccessResponse updateSchema(
       @ApiParam(value = "Name of the schema", required = true) @PathParam("schemaName") String schemaName,
-      FormDataMultiPart multiPart) {
-    return addOrUpdateSchema(schemaName, multiPart);
+      FormDataMultiPart multiPart,
+      @ApiParam(value = "Whether to force to reload segments") @DefaultValue("false") @QueryParam("forceReload") Boolean forceReload) {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java]: Can we make this parameter **boolean** instead **Boolean**? Also I would recommend name it **"reload"**.

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java [hunk 2]
```diff
@@ -178,14 +187,54 @@ private SuccessResponse addOrUpdateSchema(@Nullable String schemaName, FormDataM
...
+    }
+    successMessage.append(". Sent ").append(numReloadMessagesSent).append(" reload messages");
+  }
+
+  private List<String> getExistingTables(String schemaName) {
+    List<String> tableNamesWithType = new ArrayList<>(2);
+    ZkHelixPropertyStore<ZNRecord> propertyStore = _pinotHelixResourceManager.getPropertyStore();
+
+    // for real-time table
+    String realTimeTableName = TableNameBuilder.forType(CommonConstants.Helix.TableType.REALTIME).tableNameWithType(schemaName);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java]: TableNameBuilder.REALTIME.tableNameWithType(schemaName);

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java [hunk 3]
```diff
@@ -178,14 +187,54 @@ private SuccessResponse addOrUpdateSchema(@Nullable String schemaName, FormDataM
...
+    TableConfig tableConfig = ZKMetadataProvider.getTableConfig(propertyStore, realTimeTableName);
+    if (tableConfig != null) {
+      tableNamesWithType.add(realTimeTableName);
+    }
+
+    // for offline table
+    String offlineTableName = TableNameBuilder.forType(CommonConstants.Helix.TableType.REALTIME).tableNameWithType(schemaName);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java]: (CRITICAL) **OFFLINE**

**Comment by jackjlli** [comment]
@Jackie-Jiang It's called `auto-reload` after schema/table config updated. 
One less API call if we set reload param to true. So you can say an easy way to reload. ;)
By default it's set to false in case of backward compatibility.

**Comment by npawar** [comment]
Closed this PR due to 6 months inactivity. Reopen if needed.

**Pull request closed without merging**